apiserver/pkg/knowledgebase/knowledgebase.go

```diff
@@ -59,31 +59,49 @@ func knowledgebase2model(ctx context.Context, c client.Client, knowledgebase *v1
 		status = "Deleting"
 	}
 
+	cache := make(map[string][2]int)
 	var filegroupdetails []*generated.Filegroupdetail
-	for _, filegroupdetail := range knowledgebase.Status.FileGroupDetail {
-		var filedetails []*generated.Filedetail
-		fns := filegroupdetail.Source.Namespace
-		for _, detail := range filegroupdetail.FileDetails {
-			filedetail := &generated.Filedetail{
-				FileType:        detail.Type,
-				Count:           detail.Count,
-				Size:            detail.Size,
-				Path:            detail.Path,
-				Phase:           string(detail.Phase),
-				TimeCost:        int(detail.TimeCost),
-				UpdateTimestamp: &detail.LastUpdateTime.Time,
-			}
-			filedetails = append(filedetails, filedetail)
+	for out, fg := range knowledgebase.Spec.FileGroups {
+		groupFiles := make([]*generated.Filedetail, 0)
+		ns := knowledgebase.Namespace
+		if fg.Source.Namespace != nil {
+			ns = *fg.Source.Namespace
+		}
+		for in, fn := range fg.Paths {
+			key := fmt.Sprintf("%s/%s/%s", ns, fg.Source.Name, fn)
+			cache[key] = [2]int{out, in}
+			groupFiles = append(groupFiles, &generated.Filedetail{
+				Path:  fn,
+				Phase: string(v1alpha1.FileProcessPhasePending),
+			})
 		}
-		filegroupdetail := &generated.Filegroupdetail{
+
+		filegroupdetails = append(filegroupdetails, &generated.Filegroupdetail{
 			Source: &generated.TypedObjectReference{
-				Kind:      filegroupdetail.Source.Kind,
-				Name:      filegroupdetail.Source.Name,
-				Namespace: fns,
+				Kind:      fg.Source.Kind,
+				Name:      fg.Source.Name,
+				Namespace: fg.Source.Namespace,
 			},
-			Filedetails: filedetails,
+			Filedetails: groupFiles,
+		})
+	}
+	if len(knowledgebase.Status.FileGroupDetail) > 0 {
+		for _, filegroupdetail := range knowledgebase.Status.FileGroupDetail {
+			fns := filegroupdetail.Source.Namespace
+			for _, detail := range filegroupdetail.FileDetails {
+				key := fmt.Sprintf("%s/%s/%s", *fns, filegroupdetail.Source.Name, detail.Path)
+				v := cache[key]
+				filegroupdetails[v[0]].Filedetails[v[1]] = &generated.Filedetail{
+					FileType:        detail.Type,
+					Count:           detail.Count,
+					Size:            detail.Size,
+					Path:            detail.Path,
+					Phase:           string(detail.Phase),
+					TimeCost:        int(detail.TimeCost),
+					UpdateTimestamp: &detail.LastUpdateTime.Time,
+				}
+			}
```

Comment:
Okay, this implementation is not very good. It only allows the interface to have data, but it cannot reflect the processing status. I am considering reflecting the processing status in the status through the controller, so that the API part does not need to make this modification.

Reply:
I mean add comment so the source code :-)